lib/libuzfs/libuzfs.c

```diff
@@ -1484,10 +1470,6 @@ libuzfs_zap_update(libuzfs_dataset_handle_t *dhp, uint64_t obj, const char *key,
 	}
 
 	err = zap_update(os, obj, key, integer_size, num_integers, val, tx);
-	if (err) {
```

Comment:
if result is err, do we still need to commit the tx?

Reply:
yes, we still need dmu_tx_commit to release resources of that tx